### DISCUSSION TITLE
I'm getting a "failed to create webview" error after upgrading to 2.0 RC — what am I missing?

### DISCUSSION THREAD

**Discussion by joshsharp** [tauri-apps/tauri#10985] [Q&A]
Hello Tauri people,

I have a fairly complicated Tauri app I'm trying to migrate to v2 now it's in RC, because I need the Azure cloud signing stuff.

I ran the `migrate` command and it helped a little, but I've spent a lot of time fixing compiler issues to account for the many changes. Now that my app actually compiles, it launches with this:

> [ERROR][tauri_runtime_wry] failed to create webview: WebView2 error: WindowsError(Error { code: HRESULT(0x8007139F), message: "The group or resource is not in the correct state to perform the requested operation." })

I've seen some other issues mentioning this error that have been closed, but I'm unsure of a fix personally. It's hard to make a minimal reproducible example because I think it's a product of the migration from v1, and if I make a new test v2 project using `tauri-create-app`, it builds and runs fine. 

One difference is that my app is not a lib, but still a bin project as generated by v1, and I haven't made any of the other changes to support targeting a mobile platform. I'm not planning on ever targeting mobile so I'm assuming that's fine though.

Is there anything obvious I'm missing or should be looking for? Thanks!

**Comment by FabianLars** [comment]
> because I need the Azure cloud signing stuff.

If you're asking for signCommand then this was backported to v1 too.

> One difference is that my app is not a lib, but still a bin project as generated by v1, and I haven't made any of the other changes to support targeting a mobile platform. I'm not planning on ever targeting mobile so I'm assuming that's fine though.

That should be totally fine yes. Until very recently that was even an option in create-tauri-app and we only removed it to simplify the docs.

----

How are you creating the webviews? In tauri.conf.json or in code?
Do you have the unstable feature enabled? (Don't unless you need it for something else)
Are you setting additionalBrowserArgs for any of the windows? Can you maybe share all your window/webview configs?
Did you set a custom dataDirectory for a or all windows? (shouldn't be required but would be interesting to know what happens if you do)
Did you try to re-install / repair WebView2 already?
Can you try deleting the whole target dir (build cache)?

**Comment by joshsharp** [comment]
> If you're asking for signCommand then this was backported to v1 too.

Ah no, really? I've kept an eye on releases but I totally missed that. Cheers! I'll stick with v1 for now then and worry about upgrading later.

Thanks!
